xnmt/parameters.py

```diff
@@ -0,0 +1,60 @@
+
```

Comment:
Ok, got it! Sounds like a nice idea, maybe I'll start using this myself.. One caveat is that I think when saving the model and reloading it, this will produce unexpected results because it will set the values back to the original ones. If you want to fix that, you could do something like ```self.serialize_params['initial']=new_value``` whenever updating the value. Then, when it will be loaded the next time, the updated value will be passed as initial.

Or, if you think that's better, you can make it a bit cleaner by having both an ```initial``` and a ```current``` value, and using the second one if given. Then, it can be used for serialization by calling ```self.serialize_params['current']=new_value```

Reply:
this is a good idea. I will do and test this. BTW, does the epoch_counter persist through the separate runs?

Reply:
No, the epoch counter does not persist through runs. I think when finetuning after a pretraining phase, it makes more sense to reset the counter because we might have different data sizes etc, and it's better to define a completely new training schedule for that. But I had been thinking of making the training state persistent so that we can resume crashed trainings. That's why I added the training state here: https://github.com/neulab/xnmt/blob/master/xnmt/training_task.py#L323
But this is not implemented yet.